crates/extensions/c8y_mapper_ext/src/actor.rs

```diff
@@ -217,12 +232,68 @@ impl C8yMapperActor {
         Ok(())
     }
 
+    async fn process_upload_result(
+        &mut self,
+        cmd_id: CmdId,
+        upload_result: UploadResult,
+    ) -> Result<(), RuntimeError> {
+        match self.converter.pending_upload_operations.remove(&cmd_id) {
+            None => error!("Received an upload result for the unknown command ID: {cmd_id}"),
+            Some((smartrest_topic, clear_cmd_topic, binary_url, operation)) => {
+                let serialize_result = match operation {
+                    CumulocitySupportedOperations::C8yLogFileRequest
+                    | CumulocitySupportedOperations::C8yUploadConfigFile => self
+                        .get_smartrest_response_for_upload_result(
+                            upload_result,
+                            binary_url,
+                            operation,
+                        ),
+                    _other_types => return Ok(()), // unsupported
+                };
+
+                match serialize_result {
+                    Ok(sm_payload) => {
+                        let c8y_notification = Message::new(&smartrest_topic, sm_payload);
+                        let clear_local_cmd = Message::new(&clear_cmd_topic, "")
+                            .with_retain()
+                            .with_qos(QoS::AtLeastOnce);
```

Comment:
That's a clever and efficient use of caching info like `smartrest_topic`, `clear_cmd_topic` etc. I was expecting the `UploadResponse` handler functions in `config_operations` and `log_upload` modules respectively. But this looks fine.